modules/islandora_oadoi/islandora_oadoi.module

```diff
@@ -57,7 +57,7 @@ function islandora_oadoi_block_view($delta = '') {
             $request_url = $url . $doi;
 
             // Make the request and get results!
-            $result_json = drupal_http_request($request_url);
+            $result_json = file_get_contents($request_url);
```

Comment:
As I mentioned in Jira, this should stay [drupal_http_request](https://api.drupal.org/api/drupal/includes%21common.inc/function/drupal_http_request/7.x) you just need to account for getting an object back instead of a string.

```
$result_json = drupal_http_request($request_url);
if (!isset($result_json->error)) {
  $result_array = json_decode($result_json->data, TRUE);
  $result_free_url = $result_array['results'][0]['free_fulltext_url'];
}
```